src/wasm.h

```diff
@@ -1342,13 +1344,24 @@ class RefCast : public SpecificExpression<Expression::RefCastId> {
   Expression* rtt;
 
   void finalize();
+
+  Type getCastType();
 };
 
 class BrOnCast : public SpecificExpression<Expression::BrOnCastId> {
 public:
   BrOnCast(MixedArena& allocator) {}
 
-  void finalize() { WASM_UNREACHABLE("TODO (gc): br_on_cast"); }
+  Name name;
+  // The cast type cannot be inferred from rtt if rtt is unreachable, so we must
+  // store it explicitly.
+  Type castType;
```

Comment:
Would it make more sense to store a HeapType here rather than a Type? That way this would match the binary format better and would not have to involve unnecessary mutability information. Same question for the return type of `getCastType`.

Reply:
It could be either, in principle, yes. I tried both ways and this seems shorter in terms of the amount of code, as the other way you need to construct the Type from the HeapType more often. (Also, this matches the current BrOnExn which likewise has a Type, but I guess that's going away soon so maybe doesn't matter.)